Implemented dish flow

keren-elbaz-m opened the pull request:
added dish model, linked dishes to restaurant menus, and created routes to fetch breakfast, lunch, and dinner dishes

keren-elbaz-m committed:
Implemented dish flow

added dish model, linked dishes to restaurant menus, and created routes to fetch breakfast, lunch, and dinner dishes

DanielBaronMoveo commented on src/modules/restaurant/restaurant.controller.ts:
```diff
@@ -43,4 +43,49 @@ export class RestaurantController extends BaseController<IRestaurant> {
+                restaurantId
+            );
+            res.status(200).json({ success: true, data });
+        } catch (e) {
+            res.status(500).json({
+                success: false,
+                message: "Failed to fetch breakfast dishes",
+            });
+        }
+    };
+
+    getLunchDishes = async (req: Request, res: Response): Promise<void> => {
+        try {
+            const restaurantId = req.params.id;
+            const data = await this.restaurantService.findLunchDishes(
+                restaurantId
+            );
+            res.status(200).json({ success: true, data });
+        } catch (e) {
+            res.status(500).json({
+                success: false,
+                message: "Failed to fetch lunch dishes",
+            });
+        }
+    };
+
+    getDinnerDishes = async (req: Request, res: Response): Promise<void> => {
+        try {
+            const restaurantId = req.params.id;
+            const data = await this.restaurantService.findDinnerDishes(
+                restaurantId
+            );
+            res.status(200).json({ success: true, data });
+        } catch (e) {
+            res.status(500).json({
+                success: false,
+                message: "Failed to fetch dinner dishes",
+            });
+        }
+    };
```
That's exactly what we are trying to avoid, you are just repeating yourself.

Think of a better way to get the dishes and the "query" from a params or something.

keren-elbaz-m replied:
done

keren-elbaz-m committed:
Refactored restaurant dish menu handling

unified breakfast/lunch/dinner routes into a single route using query parameters. Replaced multiple service and repository methods with a generic getDishesByType handler.

DanielBaronMoveo commented on src/types/enums/type.enum.ts:
Change name to a more descriptive name. Something like MealTypeFilter

keren-elbaz-m replied:
changed the file to dish-time
and the enum object to DishTimeType

DanielBaronMoveo commented on src/modules/restaurant/restaurant.controller.ts:
```diff
@@ -43,4 +44,34 @@ export class RestaurantController extends BaseController<IRestaurant> {
+            const restaurantId = req.params.id;
+            const { type } = req.query;
+
+            if (!["breakfast", "lunch", "dinner"].includes(String(type))) {
```
Use the enum you created instead of hardcoded ones.

keren-elbaz-m replied:
done

DanielBaronMoveo reviewed (approved):
Minor changes, looks great! Make the changes before you merge it!

keren-elbaz-m committed:
Refactored query param enum

Renamed enum and file to DishTimeType

keren-elbaz-m committed:
Merge branch 'dev' into feature/dish-resource
